libs/community/langchain_community/vectorstores/azure_sql_server.py

```diff
@@ -0,0 +1,176 @@
+
+    """
+
+    def __init__(
+        self,
+        *,
+        connection: Optional[Connection] = None,
+        connection_string: str,
+        embedding_function: Embeddings,
+        table_name: str,
+    ) -> None:
+        """Initialize the Azure SQL Server vector store."""
+
+        self.connection_string = connection_string
+        self.embedding_function = embedding_function
+        self.table_name = table_name
+        self._bind = connection if connection else self._create_engine()
+        self.EmbeddingStore = self._get_embedding_store(table_name)
+        self._create_table_if_not_exists()
+
+    def _create_engine(self) -> Engine:
+        return create_engine(url=self.connection_string, echo=True)
+
+    def _create_table_if_not_exists(self) -> None:
+        logging.info("Creating table %s", self.table_name)
+        with Session(bind=self._bind) as session:
+            Base.metadata.create_all(session.get_bind())
+
+    def _get_embedding_store(self, name: str) -> Any:
+        global _embedding_store
+        if _embedding_store is not None:
+            return _embedding_store
+
+        class EmbeddingStore(Base):
+            """This is the base model for SQL vector store."""
+
+            __tablename__ = name
+            ID = Column(Uuid, primary_key=True, default=uuid.uuid4)
+            QUERYMETADATA = Column(JSON)
+            QUERY = Column(NVARCHAR)
+            VECTOR = Column(VARBINARY)
+
+        _embedding_store = EmbeddingStore
+        return _embedding_store
+
+    @property
+    def embeddings(self) -> Embeddings:
+        return self.embedding_function
+
+    @classmethod
+    def from_texts(
+        cls: type[VST],
+        texts: List[str],
+        embedding: Embeddings,
+        metadatas: List[dict] | None = None,
+        **kwargs: Any,
+    ) -> VST:
+        return super().from_texts(texts, embedding, metadatas, **kwargs)
+
+    def similarity_search(
+        self, query: str, k: int = 4, **kwargs: Any
+    ) -> List[Document]:
+        return super().similarity_search(query, k, **kwargs)
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Compute the embeddings for the input texts and store embeddings
+        in the vectorstore.
+
+        Args:
+            texts: Iterable of strings to add into the vectorstore.
+            metadatas: List of metadatas (python dicts) associated with the input texts.
+            ids: List of IDs for the input texts.
+            kwargs: vectorstore specific parameters.
+
+        Returns:
+            List of IDs generated from adding the texts into the vectorstore.
+        """
+
+        # Embed the texts passed in.
+        embedded_texts = self.embedding_function.embed_documents(
+            list(texts)
+        )  # List[List[float]]
```

Comment:
Why is the comment #List[List[float]] ? does it represent the type of embedded_texts?

Reply:
It the return type from the embed_documents function. I'll remove the comment, it doesn't help out there.